test/commons/dom/url-props-from-attribute.js

```diff
@@ -0,0 +1,168 @@
+describe('dom.urlPropsFromAttribute', function () {
+	'use strict';
+
+	var fixture = document.getElementById('fixture');
+	var queryFixture = axe.testUtils.queryFixture;
+
+	afterEach(function () {
+		fixture.innerHTML = '';
+	});
+
+	it('returns undefined when given node does not have specified attribute', function () {
+		var vNode = queryFixture(
+			'<button id="target" role="link">Schedule appointment</button>'
+		);
+		var actual = axe.commons.dom.urlPropsFromAttribute(vNode.actualNode, 'href');
+		assert.isUndefined(actual);
+	});
+
+	it('returns undefined when `A` has no `HREF` attribute', function () {
+		var vNode = queryFixture('<a id="target">Follow us on Instagram</a>');
+		var actual = axe.commons.dom.urlPropsFromAttribute(vNode.actualNode, 'href');
+		assert.isUndefined(actual);
+	});
+
+	it('returns URL properties when `A` has empty `HREF`', function () {
+		var vNode = queryFixture(
+			'<a id="target" href="">Follow us on Instagram</a>'
+		);
+		var expected = {
```

Comment:
Please provide a URL for this. This will break if we run this on a different port, or use an IP address instead of localhost.